File: server/routes/history.py
from __future__ import annotations

from fastapi import APIRouter, Query

from server.dependencies import require_workspace
from server.schemas import HistorySearchResponse
from services.history_service import search_history

router = APIRouter(prefix="/api/history", tags=["history"])


@router.get("/search", response_model=HistorySearchResponse)
async def search_history_endpoint(
    q: str = Query(default=""),
    offset: int = Query(default=0, ge=0),
    limit: int = Query(default=10, ge=1, le=100),
):
    require_workspace()
    return search_history(q, offset=offset, limit=limit)

File: frontend/src/api/history.ts
import { request } from "./core";
import type { HistorySearchResponse } from "../types";

export function searchHistory(
  query: string,
  offset: number = 0,
  limit: number = 10,
): Promise<HistorySearchResponse> {
  const params = new URLSearchParams();
  params.set("q", query);
  params.set("offset", String(offset));
  params.set("limit", String(limit));
  return request<HistorySearchResponse>(`/history/search?${params.toString()}`);
}
